src/web/core/src/main/java/org/geoserver/web/data/resource/BasicResourceConfig.java

```diff
@@ -457,4 +326,130 @@ public void validate(final Form<?> form) {
             }
         }
     }
+
+    public boolean addOtherSRS(ResourceInfo resourceInfo) {
+
+        // first check if its WFS-NG or WMSStore
+        List<String> otherSRS = getOtherSRSFromWFSOrWMS(resourceInfo);
+
+        if (otherSRS != null) {
+            if (!otherSRS.isEmpty()) {
+                resourceInfo
+                        .getMetadata()
+                        .put(FeatureTypeInfo.OTHER_SRS, String.join(",", otherSRS));
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String getActualNativeSRSCode(ResourceInfo resourceInfo) {
+
+        try {
+            if (resourceInfo instanceof FeatureTypeInfo) {
+                FeatureTypeInfo ft = (FeatureTypeInfo) resourceInfo;
+                return "EPSG:"
+                        + CRS.lookupEpsgCode(
+                                ft.getFeatureSource(null, null).getInfo().getCRS(), false);
+            } else {
+                CatalogBuilder cb = new CatalogBuilder(GeoServerApplication.get().getCatalog());
+                cb.setStore(resourceInfo.getStore());
+                ReferencedEnvelope bounds = cb.getNativeBounds(resourceInfo);
+                return "EPSG:" + CRS.lookupEpsgCode(bounds.getCoordinateReferenceSystem(), false);
+            }
+
+        } catch (Exception e) {
+            LOGGER.log(
+                    Level.SEVERE,
+                    "Error getting actual Native SRS code for resource "
+                            + resourceInfo.getNativeName(),
+                    e);
+        }
+
+        return "";
+    }
+
+    private List<String> getOtherSRSFromWFSOrWMS(ResourceInfo resourceInfo) {
+        // first check if its WFS-NG
+        List<String> otherSRS = DataStoreUtils.getOtherSRSFromWfsNg(resourceInfo);
+        // second check if its wms
+        if (otherSRS.isEmpty()) otherSRS = DataStoreUtils.getOtherSRSFromWMSStore(resourceInfo);
+
+        return otherSRS;
+    }
+
+    private CRSPanel getSelectableNativeCRSPanel(
```

Comment:
I would add a comment describing what's this about, new panel for cascaded stores?

Reply:
will add